### PR TITLE
fix: sequencer stops posting batches after hub disconnection

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close #XXX

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/production_test.go [hunk 1]
```diff
@@ -167,64 +165,6 @@ func TestCreateEmptyBlocksNew(t *testing.T) {
 	assert.True(foundTx)
 }
 
-func TestBatchSubmissionAfterTimeout(t *testing.T) {
```
Comment (mtsitrin) [inline block/production_test.go:170]: moved to submit_test

File: block/submit.go [hunk 2]
```diff
@@ -14,63 +14,60 @@ func (m *Manager) SubmitLoop(ctx context.Context) {
...
+	}
+
+	// Submit batch if we've reached the batch size and there isn't another batch currently in submission process.
+	if m.batchInProcess.Load() == true {
+		m.logger.Debug("Batch submission already in process, skipping submission")
+		return
+	}
+
+	m.batchInProcess.Store(true)
+	defer func() {
```
Comment (mtsitrin) [inline block/submit.go:46]: this is the fix

File: block/submit.go [hunk 3]
```diff
@@ -14,63 +14,60 @@ func (m *Manager) SubmitLoop(ctx context.Context) {
...
+		m.batchInProcess.Store(false)
+	}()
 
-	isLastBlockEmpty, err := m.validateLastBlockInBatchIsEmpty(startHeight, endHeight)
```
Comment (mtsitrin) [inline block/submit.go:64]: moved down, to use `actualEndHeight` instead of `endHeight`

File: block/submit.go [hunk 4]
```diff
@@ -93,9 +102,6 @@ func (m *Manager) submitNextBatch(ctx context.Context) (uint64, error) {
 	}
 
 	// Submit batch to SL
-	// TODO(omritoptix): Handle a case where the SL submission fails due to syncTarget out of sync with the latestHeight in the SL.
```
Comment (mtsitrin) [inline block/submit.go:96]: removed this todo, as we're in `aggregator` mode, and expect our state to be source of truth.

**Review by omritoptix — APPROVED** [review]

**Pull request merged**